pkg/util/pem.go

```diff
@@ -121,3 +104,29 @@ func DecodeX509CertificateChainBytes(certBytes []byte) ([]*x509.Certificate, err
 
 	return certs, nil
 }
+
```

Comment:
I think the following functions should have `CertPool` as receiver. Why not?

Reply:
Also, we could rename the functions as follows:
- GetSplitPEMBundle -> AsSplitPEMBundle
- GetSplitPEMBundleBytes -> AsPEMBundleBytes
- GetSplitPEMBundleStrings-> AsPEMBundleStrings
- GetCertsList -> AsCertificateList

Reply:
Functions were renamed.
Regarding `CertPool`  as argument for DecodeX509CertificateChainBytes() - I not agree.
Let me explain: an idea here is `PEM` is responsible for providing prepared data for consumers, `CertPool` is responsible for store, manage, filter, deduplicate, etc of `Certificates`, it provides data for consumers like `PEM`. 
Imagine, we want to provide `DER` encoded data instead of `PEM` one. 
Moreover, I'm thinking about making `CertPool` abstract structure (interface).

Reply:
> Regarding CertPool as argument for DecodeX509CertificateChainBytes()

I think Erik was referring to the functions below (AsSplitPEMBundle, AsPEMBundleBytes, AsCertificateList, GetCertificatesQuantity) rather than  DecodeX509CertificateChainBytes.

> An idea here is `PEM` is responsible for providing prepared data for consumers in form that is needed by consumers, consumers by themself shouldn't know how, where data was taken from, CertPool is responsible for store, manage, filter, de-duplicate, etc of Certificates, it provides data for `PEM`, so end consumers shouldn't know how CertPool works inside (encapsulation).

I'm not sure what `PEM` refers to. Is it a type that will be created in the future and that, for now, exists through the existence of the AsPEM functions?

> Imagine, we want to provide `DER` encoded data instead of PEM one.

I imagine that CertPool would just need to add a couple of new receiver functions:

```go
func (*CertPool) AsDER() [][]byte
```

Why not? I don't see the benefits of encapsulating the "export" (DER or PEM) logic.

> Moreover, I'm thinking about making CertPool abstract structure (interface).

What would be the benefits of turning this into an interface? Would that be for testing purposes?